### PR TITLE
Partially fix EagerUniqueKeyTest

### PR DESCRIPTION
**Pull request by geoand**
The custom cleanup is done because the actual test itself passes, what fails before this change is the loading of entities which occurs when the database is being cleared during test cleanup.

Furthermore, fix erroneous assertion in `EagerUniqueKeyTest`.

After this change, only `testMergeReference` fails (whereas before all tests failed) with:

```
java.util.concurrent.CompletionException: org.hibernate.HibernateException: java.util.concurrent.CompletionException: org.hibernate.AssertionFailure: non-transient entity has a null id: org.hibernate.reactive.EagerUniqueKeyTest$Bar$HibernateProxy$scYAEL1t
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:315)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:320)
	at java.base/java.util.concurrent.CompletableFuture.uniHandle(CompletableFuture.java:936)
	at java.base/java.util.concurrent.CompletableFuture.uniHandleStage(CompletableFuture.java:950)
	at java.base/java.util.concurrent.CompletableFuture.handle(CompletableFuture.java:2340)
	at java.base/java.util.concurrent.CompletableFuture.handle(CompletableFuture.java:144)
	at org.hibernate.reactive.session.impl.ReactiveSessionImpl.fireMerge(ReactiveSessionImpl.java:1010)
	at org.hibernate.reactive.session.impl.ReactiveSessionImpl.reactiveMerge(ReactiveSessionImpl.java:993)
	at org.hibernate.reactive.stage.impl.StageSessionImpl.merge(StageSessionImpl.java:174)
	at org.hibernate.reactive.EagerUniqueKeyTest.lambda$testMergeReference$14(EagerUniqueKeyTest.java:78)
	at org.hibernate.reactive.stage.Stage$SessionFactory.lambda$withTransaction$0(Stage.java:2619)
	at org.hibernate.reactive.stage.impl.StageSessionFactoryImpl.lambda$withTransaction$19(StageSessionFactoryImpl.java:233)
	at org.hibernate.reactive.stage.impl.StageSessionImpl$Transaction.executeInTransaction(StageSessionImpl.java:389)
	at org.hibernate.reactive.stage.impl.StageSessionImpl$Transaction.lambda$execute$0(StageSessionImpl.java:379)
	at java.base/java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:1150)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2147)
	at io.vertx.core.Future.lambda$toCompletionStage$3(Future.java:384)
	at io.vertx.core.impl.future.FutureImpl$3.onSuccess(FutureImpl.java:141)
	at io.vertx.core.impl.future.FutureImpl$ListenerArray.onSuccess(FutureImpl.java:262)
	at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
	at io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:211)
	at io.vertx.core.impl.future.PromiseImpl.tryComplete(PromiseImpl.java:23)
	at io.vertx.core.impl.future.PromiseImpl.onSuccess(PromiseImpl.java:49)
	at io.vertx.core.impl.future.PromiseImpl.handle(PromiseImpl.java:41)
	at io.vertx.sqlclient.impl.TransactionImpl.lambda$wrap$0(TransactionImpl.java:72)
	at io.vertx.core.impl.future.FutureImpl$3.onSuccess(FutureImpl.java:141)
	at io.vertx.core.impl.future.FutureBase.lambda$emitSuccess$0(FutureBase.java:54)
	at io.vertx.core.impl.EventLoopContext.execute(EventLoopContext.java:86)
	at io.vertx.core.impl.ContextBase.execute(ContextBase.java:225)
	at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:51)
	at io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:211)
	at io.vertx.core.impl.future.PromiseImpl.tryComplete(PromiseImpl.java:23)
	at io.vertx.core.impl.future.PromiseImpl.onSuccess(PromiseImpl.java:49)
	at io.vertx.core.impl.future.PromiseImpl.handle(PromiseImpl.java:41)
	at io.vertx.core.impl.future.PromiseImpl.handle(PromiseImpl.java:23)
	at io.vertx.pgclient.impl.PgSocketConnection.lambda$doSchedule$2(PgSocketConnection.java:155)
	at io.vertx.sqlclient.impl.command.CommandResponse.fire(CommandResponse.java:46)
	at io.vertx.sqlclient.impl.SocketConnectionBase.handleMessage(SocketConnectionBase.java:292)
	at io.vertx.pgclient.impl.PgSocketConnection.handleMessage(PgSocketConnection.java:97)
	at io.vertx.sqlclient.impl.SocketConnectionBase.lambda$init$0(SocketConnectionBase.java:105)
	at io.vertx.core.impl.EventLoopContext.emit(EventLoopContext.java:55)
	at io.vertx.core.impl.ContextBase.emit(ContextBase.java:239)
	at io.vertx.core.net.impl.NetSocketImpl.handleMessage(NetSocketImpl.java:390)
	at io.vertx.core.net.impl.ConnectionBase.read(ConnectionBase.java:157)
	at io.vertx.core.net.impl.VertxHandler.channelRead(VertxHandler.java:153)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireChannelRead(CombinedChannelDuplexHandler.java:436)
	at io.vertx.pgclient.impl.codec.PgEncoder.lambda$write$0(PgEncoder.java:98)
	at io.vertx.pgclient.impl.codec.PgCommandCodec.handleReadyForQuery(PgCommandCodec.java:139)
	at io.vertx.pgclient.impl.codec.PgDecoder.decodeReadyForQuery(PgDecoder.java:237)
	at io.vertx.pgclient.impl.codec.PgDecoder.channelRead(PgDecoder.java:96)
	at io.netty.channel.CombinedChannelDuplexHandler.channelRead(CombinedChannelDuplexHandler.java:251)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: org.hibernate.HibernateException: java.util.concurrent.CompletionException: org.hibernate.AssertionFailure: non-transient entity has a null id: org.hibernate.reactive.EagerUniqueKeyTest$Bar$HibernateProxy$scYAEL1t
	at app//org.hibernate.reactive.session.impl.ReactiveExceptionConverter.convert(ReactiveExceptionConverter.java:31)
	at app//org.hibernate.reactive.session.impl.ReactiveSessionImpl.lambda$fireMerge$23(ReactiveSessionImpl.java:1021)
	at java.base@17.0.6/java.util.concurrent.CompletableFuture.uniHandle(CompletableFuture.java:934)
	... 68 more
Caused by: java.util.concurrent.CompletionException: org.hibernate.AssertionFailure: non-transient entity has a null id: org.hibernate.reactive.EagerUniqueKeyTest$Bar$HibernateProxy$scYAEL1t
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:315)
	at java.base/java.util.concurrent.CompletableFuture.uniComposeStage(CompletableFuture.java:1194)
	at java.base/java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:2309)
	at java.base/java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:144)
	at org.hibernate.reactive.engine.impl.EntityTypes.resolveIdOrUniqueKey(EntityTypes.java:301)
	at org.hibernate.reactive.engine.impl.EntityTypes.replace(EntityTypes.java:289)
	at org.hibernate.reactive.engine.impl.EntityTypes.replace(EntityTypes.java:240)
	at org.hibernate.reactive.engine.impl.EntityTypes.lambda$replace$7(EntityTypes.java:212)
	at org.hibernate.reactive.util.impl.CompletionStages.lambda$loop$7(CompletionStages.java:373)
	at org.hibernate.reactive.util.impl.CompletionStages$ArrayLoop.next(CompletionStages.java:443)
	at org.hibernate.reactive.util.async.impl.AsyncTrampoline.lambda$asyncWhile$1(AsyncTrampoline.java:215)
	at org.hibernate.reactive.util.async.impl.AsyncTrampoline$TrampolineInternal.unroll(AsyncTrampoline.java:121)
	at org.hibernate.reactive.util.async.impl.AsyncTrampoline$TrampolineInternal.trampoline(AsyncTrampoline.java:102)
	at org.hibernate.reactive.util.async.impl.AsyncTrampoline.asyncWhile(AsyncTrampoline.java:197)
	at org.hibernate.reactive.util.async.impl.AsyncTrampoline.asyncWhile(AsyncTrampoline.java:215)
	at org.hibernate.reactive.util.impl.CompletionStages.loop(CompletionStages.java:374)
	at org.hibernate.reactive.engine.impl.EntityTypes.replace(EntityTypes.java:209)
	at org.hibernate.reactive.event.impl.DefaultReactiveMergeEventListener.copyValues(DefaultReactiveMergeEventListener.java:570)
	at org.hibernate.reactive.event.impl.DefaultReactiveMergeEventListener.lambda$entityIsTransient$4(DefaultReactiveMergeEventListener.java:266)
	at java.base/java.util.concurrent.CompletableFuture.uniComposeStage(CompletableFuture.java:1187)
	at java.base/java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:2309)
	at java.base/java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:144)
	at org.hibernate.reactive.event.impl.DefaultReactiveMergeEventListener.entityIsTransient(DefaultReactiveMergeEventListener.java:266)
	at org.hibernate.reactive.event.impl.DefaultReactiveMergeEventListener.reactiveOnMerge(DefaultReactiveMergeEventListener.java:202)
	at org.hibernate.reactive.event.impl.DefaultReactiveMergeEventListener.reactiveOnMerge(DefaultReactiveMergeEventListener.java:100)
	at org.hibernate.event.service.internal.EventListenerGroupImpl.lambda$fireEventOnEachListener$0(EventListenerGroupImpl.java:153)
	at java.base/java.util.concurrent.CompletableFuture.uniComposeStage(CompletableFuture.java:1187)
	at java.base/java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:2309)
	at java.base/java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:144)
	at org.hibernate.event.service.internal.EventListenerGroupImpl.fireEventOnEachListener(EventListenerGroupImpl.java:153)
	at org.hibernate.reactive.session.impl.ReactiveSessionImpl.fireMerge(ReactiveSessionImpl.java:1009)
	... 64 more
Caused by: org.hibernate.AssertionFailure: non-transient entity has a null id: org.hibernate.reactive.EagerUniqueKeyTest$Bar$HibernateProxy$scYAEL1t
	at app//org.hibernate.reactive.engine.impl.EntityTypes.lambda$resolveIdOrUniqueKey$11(EntityTypes.java:303)
	at java.base@17.0.6/java.util.concurrent.CompletableFuture.uniComposeStage(CompletableFuture.java:1187)
	... 93 more
```

### CODE REVIEW

File: hibernate-reactive-core/src/test/java/org/hibernate/reactive/EagerUniqueKeyTest.java [hunk 1]
```diff
@@ -56,8 +64,8 @@ public void testMergeDetached(TestContext context) {
 				.thenCompose( i -> getSessionFactory()
 						.withTransaction( session -> session.merge( new Foo( bar ) ) ) )
 				.thenCompose( result -> getSessionFactory()
-						.withTransaction( session -> session.fetch( result.bar )
-						.thenAccept( b -> context.assertEquals( "unique2", b.key ) )
+						.withTransaction( session -> session.fetch( result.getBar() )
+						.thenAccept( b -> context.assertEquals( Bar.class.getName(), b.getClass().getName() ) )
```
Comment (DavideD) [inline hibernate-reactive-core/src/test/java/org/hibernate/reactive/EagerUniqueKeyTest.java:68]: Did you change the assertion on purpose?

Shouldn't it be
```
.thenAccept( b -> context.assertEquals( "unique2", b.getKey() ) )?
```
?

Comment (geoand) [reply hibernate-reactive-core/src/test/java/org/hibernate/reactive/EagerUniqueKeyTest.java:68]: Good point!

Comment (geoand) [reply hibernate-reactive-core/src/test/java/org/hibernate/reactive/EagerUniqueKeyTest.java]: Done

**Comment by DavideD** [comment]
Thanks, merged

**Comment by geoand** [comment]
👍🏼

**Pull request merged**